tests/integration/test_log.py

```diff
@@ -258,3 +258,31 @@ def test_log_partial_entry(cluster):
     # There will be 9 valid entries + 1 NOOP entry = 10
     assert cluster.node(1).info()['raft_log_entries'] == 10
     assert cluster.execute('get', 'x') == ('b' * 15000).encode('utf-8')
+
+
+def test_log_corrupt_entry(cluster):
```

Comment:
What about corruption in the middle of the file (not the in last entry/ header) or garbage "valid" last entry?

Reply:
I can add a test, but it will behave the same, prune it to that point.

Reply:
right, but it will test that indeed we prune all the irrelevant entries

Reply:
modified the test to add a test case for this